src/it/resources-noncompilable/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputFormattedWhitespaceAroundWhen.java

```diff
@@ -0,0 +1,91 @@
+// non-compiled with javac: Compilable with Java21
+
+package com.google.checkstyle.test.chapter4formatting.rule462horizontalwhitespace;
+
+class InputFormattedWhitespaceAroundWhen {
+
+  /** method. */
+  void test(Object o) {
+    switch (o) {
+      case Integer i when (i == 0) -> {}
+      // 2 violations above:
+      //  ''{' is not followed by whitespace.'
+      //  ''}' is not preceded with whitespace.'
```

Comment:
@Zopsss, until what issue this is a violation ?

Reply:
https://github.com/checkstyle/checkstyle/issues/15338